### ISSUE TITLE
Does sleeping over night work?

### ISSUE THREAD

**Issue by edwardhorsford** [jeroen85/epaper-display#2]
Firstly, thank you so much for sharing this repo - it gave me a massive help in configuring my display/esp.

Having gone through the code I wonder if the nighttime sleeping is working as expected.

The ESP is sleeping most of the time anyway, and my hunch is that it will likely be asleep when the scheduled night-sleep is meant to commence - and so it never triggers. Some quick testing of mine has suggested if the esp is asleep during the scheduled time it won't wake up - it'll continue as normal.

**Comment by jeroen85** [comment]
Thanks for your nice words.

In my experience, it works.

In my case it does a last update a little bit after 21:00. Then it goes to sleep for 60 minutes again to wake up around 22:0x.

During start-up, it fetches time from Home Assistant and checks if the hours are equal to 22. It does not check the minutes or seconds.
As the hours are now 22, it will enter a night sleep.

```yaml
time:
  - platform: homeassistant
    id: esptime
    timezone: Europe/Amsterdam
    on_time_sync:
      - then:
          - logger.log: Time has been set and is valid!       
    on_time:
      # Every evening after 22:00
      - hours: 22
        then:
          - logger.log: Going for a sleep during the night. Good night!   
          - deep_sleep.enter:
              id: deep_sleep_1
              until: "6:00:00"
              time_id: esptime
```

See below the connection status in Home Assistant. 
![image](https://github.com/jeroen85/epaper-display/assets/26403565/67553864-d75b-4ed1-a5ac-31373cfe44f7)

**Comment by edwardhorsford** [comment]
Sorry for the false report! I'd misinterpreted the `hours: 22` as being for 22:00 and in testing had set it explicitly with a minute as I was testing past the hour - which was clearly the reason this didn't trigger.

I wrote a different solution that ends up with a similar thing - the regular sleep script checks if the hour is x, and if so goes to sleep until time y.

One query - mine was set to wake at 8:00 this morning, but I can see in the logs it woke at 7:55. Similarly, I see yours is set to wake at 6am, but your logs indicate it woke up 3 minutes early. Is it expected that it wake up slightly early?

**Comment by jeroen85** [comment]
No problem. And good suggestion with respect to your script.

I don't know why it wakes up a bit early. I guess the deep sleep timer in the cpu is not very accurate..?

**Comment by edwardhorsford** [comment]
Yeah I've decided it must just be the crystal accuracy - as a hack I've added 1% to my intended sleep times.

I think your script just sleeps for 60 mins from when it was booted so it doesn't matter much - mine instead attempts to sync with the start of the hour so it wakes up as new forecasts come in. But because of the inaccuracy it was actually waking up 30s early, then going to sleep for 30 seconds and waking up again - doubling the wakes. I've put in some extra checks now to stop this.

Will close this issue as I was mistaken anyway.
